src/PhoneNumber/PhoneNumber.js

```diff
@@ -27,27 +29,66 @@ class PhoneNumber extends Component {
     countries: [],
     disabled: false,
     floatingLabel: true,
+    formattedNumber: NO_VALUE,
     onBlur: () => {},
     onChange: () => {},
     onFocus: () => {},
-    prefix: '',
+    prefix: NO_VALUE,
     readOnly: false
   };
 
   constructor(props) {
     super(props);
     this.numberInputRef = React.createRef();
 
-    const { value = NO_VALUE, prefix } = this.props;
-
+    const { formattedNumber, prefix, value = NO_VALUE } = this.props;
     const phoneNumber = value;
 
     this.state = {
-      prefix: prefix,
+      formattedNumber,
       phoneNumber,
+      prefix,
       isFocused: false
     };
   }
+  componentWillMount() {
```

Comment:
componentWillMount is deprecated in newer versions of react. Also it seems that the logic inside it is just to initialize the state. I think we can have all the logic directly within the constructor